module/VuFind/src/VuFind/Search/ProQuestFSG/Results.php

```diff
@@ -0,0 +1,175 @@
+
+    /**
+     * Simplified version of result facets.
+     *
+     * @var array
+     */
+    protected $simplifiedResponseFacets = [];
+
+    /**
+     * Support method for performAndProcessSearch -- perform a search based on the
+     * parameters passed to the object.
+     *
+     * @return void
+     */
+    protected function performSearch()
+    {
+        $query  = $this->getParams()->getQuery();
+        $allTerms = $query->getAllTerms();
+        if (!strlen($allTerms)) {
+            $this->storeErrorResponse('empty_search_disallowed');
+            return;
+        }
+        $limit  = $this->getParams()->getLimit();
+        $offset = $this->getStartRecord();
+        $params = $this->getParams()->getBackendParameters();
+        $command = new SearchCommand(
+            $this->backendId,
+            $query,
+            $offset,
+            $limit,
+            $params
+        );
+        $collection = $this->getSearchService()
+            ->invoke($command)->getResult();
+
+        $this->resultTotal = $collection->getTotal();
+        $this->results = $collection->getRecords();
+
+        // ProQuest does not return facets unless the startRecord is 1
+        if ($offset === 1) {
+            $this->responseFacets = $collection->getFacets();
+            $this->simplifiedResponseFacets = $this->simplifyFacets($this->responseFacets);
+        }
+    }
+
+    /**
+     * Store an empty response with an error message instead of performing a search.
+     *
+     * @param string|array $error Error message(s) to display to user.
+     *
+     * @return void
+     */
+    protected function storeErrorResponse(string|array $error): void
+    {
+        $this->resultTotal = 0;
+        $this->results = [];
+        $this->simplifiedResponseFacets = [];
+        $this->responseFacets = [];
+        $this->errors = (array)$error;
+    }
+
+    /**
+     * Returns the stored list of facets for the last search
+     *
+     * @param array $filter Array of field => on-screen description listing
+     * all of the desired facet fields; set to null to get all configured values.
+     *
+     * @return array        Facets data arrays
+     */
+    public function getFacetList($filter = null)
+    {
+        if (!empty($this->getParams()->getFacetConfig()) && empty($this->simplifiedResponseFacets)) {
+            // Save actual search data
+            $resultTotal = $this->resultTotal;
+            $results = $this->results;
+            $suggestions = $this->suggestions;
+            $errors = $this->errors;
+
+            // The API only provides facets when startRecord == 1.
+            $this->overrideStartRecord(1);
+            $this->performAndProcessSearch();
+
+            // Restore actual search data
+            $this->resultTotal = $resultTotal;
+            $this->results = $results;
+            $this->suggestions = $suggestions;
+            $this->errors = $errors;
+            $this->overrideStartRecord(null);
```

Comment:
While I don't know how or why this would happen in practice, for the sake of consistency, maybe it would be best to store the current value of `$this->startRecordOverride` with the other backed-up properties above, and then restore that value here, just in case it's already something non-null for some weird reason.

Reply:
Agreed.  This is done.